src/API/Management/ClientGrants.php

```diff
@@ -2,75 +2,193 @@
+    {
+        if (empty($audience) || ! is_string($audience)) {
+            throw new CoreException('Empty or invalid "audience" parameter.');
+        }
+
+        return $this->getAll(['audience' => $audience], $page, $per_page);
+    }
+
+    /**
+     * Get Client Grants by Client ID.
+     * Required scope: "read:client_grants"
+     *
+     * @param string       $client_id Client ID to filter by.
+     * @param null|integer $page      The page number, zero based.
+     * @param null|integer $per_page  The amount of entries per page.
+     *
+     * @return mixed
+     *
+     * @throws CoreException Thrown when $client_id is empty or not a string.
+     * @throws \Exception Thrown by the Guzzle HTTP client when there is a problem with the API call.
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Client_Grants/get_client_grants
+     */
+    public function getByClientId($client_id, $page = null, $per_page = null)
+    {
+        if (empty($client_id) || ! is_string($client_id)) {
+            throw new CoreException('Empty or invalid "client_id" parameter.');
+        }
+
+        return $this->getAll(['client_id' => $client_id], $page, $per_page);
+    }
+
+    /**
+     * Create a new Client Grant.
+     * Required scope: "create:client_grants"
+     *
+     * @param string $client_id Client ID to receive the grant.
+     * @param string $audience  Audience identifier for the API being granted.
+     * @param array  $scope     Array of scopes for the grant.
+     *
+     * @return mixed
+     *
+     * @throws CoreException Thrown when $client_id or $audience are empty or not a string.
+     * @throws \Exception Thrown by the Guzzle HTTP client when there is a problem with the API call.
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Client_Grants/post_client_grants
+     */
+    public function create($client_id, $audience, array $scope = [])
+    {
+        if (empty($client_id) || ! is_string($client_id)) {
+            throw new CoreException('Empty or invalid "client_id" parameter.');
+        }
+
+        if (empty($audience) || ! is_string($audience)) {
+            throw new CoreException('Empty or invalid "audience" parameter.');
+        }
+
+        return $this->apiClient->method('post')
+            ->addPath('client-grants')
+            ->withBody(json_encode([
+                'client_id' => $client_id,
+                'audience' => $audience,
+                'scope' => $scope,
+            ]))
+            ->call();
     }
 
     /**
+     * Delete a Client Grant by ID.
+     * Required scope: "delete:client_grants"
+     *
+     * @param string $id Client Grant ID to delete.
      *
-     * @param  string $client_id
-     * @param  string $audience
-     * @param  string $scope
      * @return mixed
+     *
+     * @throws \Exception Thrown by the Guzzle HTTP client when there is a problem with the API call.
```

Comment:
Although we use Guzzle I would not mention it by name, goes for all mentions in this PR.

Reply:
We use it in the variable name for the API. It will also appear in the error name. I think it's helpful here in case someone is getting a fatal error and want to know where to start troubleshooting.